google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/MessageDispatcher.java

```diff
@@ -128,12 +100,22 @@ public String toString() {
     private final int outstandingBytes;
     private final long receivedTimeMillis;
     private final Instant totalExpiration;
+    private final SettableApiFuture<AckResponse> messageFuture;
 
-    private AckHandler(String ackId, int outstandingBytes, Instant totalExpiration) {
+    private AckHandler(
+        String ackId,
+        int outstandingBytes,
+        Instant totalExpiration,
+        SettableApiFuture<AckResponse> messageFuture) {
```

Comment:
nit. Calling this messageFuture is a little misleading since we are not waiting for a message, we are waiting for the ack/nack to complete. Maybe call it ackResponseFuture or similar?

Reply:
Good call, will update.

Reply:
This is the `messageFuture` that will be sent to the users, it is here to be passed through to the `StreamingSubscriberConnection.sendAckOperations`